contracts/EventBasedPredictionMarket.sol

```diff
@@ -0,0 +1,318 @@
+        require(identifier == priceIdentifier, "same identifier");
+        require(keccak256(ancillaryData) == keccak256(customAncillaryData), "same ancillary data");
+        require(refund == proposerReward, "same proposerReward amount");
+
+        _requestOraclePrice();
+    }
+
+    /****************************************
+     *          POSITION FUNCTIONS          *
+     ****************************************/
+
+    /**
+     * @notice Creates a pair of long and short tokens equal in number to tokensToCreate. Pulls the required collateral
+     * amount into this contract, defined by the collateralPerPair value.
+     * @param tokensToCreate number of long and short synthetic tokens to create.
+     * @return collateralUsed total collateral used to mint the synthetics.
+     */
+    function create(uint256 tokensToCreate) public requestInitialized returns (uint256 collateralUsed) {
+        // Note the use of multiply and ceiling to prevent small collateralPerPair causing rounding of collateralUsed to 0 enabling
+        // callers to mint dust LSP tokens without paying any collateral.
+        uint256 mulRaw = tokensToCreate * collateralPerPair;
+        uint256 mulFloor = mulRaw / 1e18;
+        uint256 mod = mulRaw % 1e18;
+        collateralUsed = mod != 0 ? mulFloor + 1 : mulFloor; // ceil(mulRaw / 1e18)
+
+        collateralToken.safeTransferFrom(msg.sender, address(this), collateralUsed);
+
+        require(longToken.mint(msg.sender, tokensToCreate));
+        require(shortToken.mint(msg.sender, tokensToCreate));
+
+        emit TokensCreated(msg.sender, collateralUsed, tokensToCreate);
+    }
+
+    /**
+     * @notice Redeems a pair of long and short tokens equal in number to tokensToRedeem. Returns the commensurate
+     * amount of collateral to the caller for the pair of tokens, defined by the collateralPerPair value.
+     * @param tokensToRedeem number of long and short synthetic tokens to redeem.
+     * @return collateralReturned total collateral returned in exchange for the pair of synthetics.
+     */
+    function redeem(uint256 tokensToRedeem) public returns (uint256 collateralReturned) {
+        require(longToken.burnFrom(msg.sender, tokensToRedeem));
+        require(shortToken.burnFrom(msg.sender, tokensToRedeem));
+
+        collateralReturned = (tokensToRedeem * collateralPerPair) / 1e18;
+
+        collateralToken.safeTransfer(msg.sender, collateralReturned);
+
+        emit TokensRedeemed(msg.sender, collateralReturned, tokensToRedeem);
+    }
+
+    /**
+     * @notice Settle long and/or short tokens in for collateral at a rate informed by the contract settlement.
+     * @param longTokensToRedeem number of long tokens to settle.
+     * @param shortTokensToRedeem number of short tokens to settle.
+     * @return collateralReturned total collateral returned in exchange for the pair of synthetics.
+     */
+    function settle(uint256 longTokensToRedeem, uint256 shortTokensToRedeem)
```

Comment:
I know it's pretty unnecessary, since you can just have the first person that `settles` also call `getExpirationPrice()`, but thoughts on using the settled price callback to get the settlement price instead of having it be initiated here? Outcome is basically the same, just seems like a good thing to showcase.

Reply:
Yes! This would simplify the contract even more, good idea.